lang/image.bzl

```diff
@@ -34,8 +34,8 @@ def _binary_name(ctx):
     # /app/foo/bar/baz/blah
     return "/".join([
         ctx.attr.directory,
-        ctx.attr.binary.label.package,
-        ctx.attr.binary.label.name,
+        ctx.attr.binary[0].label.package,
```

Comment:
Why does this change to only specify the first element/object in `ctx.attr.binary` need to be made?

Reply:
When `cfg` is set to a transition configuration, we have what is called an "Outgoing Edge Transition" (https://docs.bazel.build/versions/master/skylark/config.html#outgoing-edge-transitions).  Outgoing edge transitions are 1:1 or 1:2+: you can basically say "I want this dependency both for Linux, for iOS, and for Mac OSX".  So the attribute is sneakily changed to an array.  The order of what you get in this array is defined by the transition.